asapdiscovery-data/asapdiscovery/data/schema_v2/target.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple, Union  # noqa: F401
+
+from asapdiscovery.data.openeye import oechem, oemol_to_pdb_string, pdb_string_to_oemol
+from pydantic import Field
+
+from .dynamic_properties import TargetType
+from .schema_base import (
+    DataModelAbstractBase,
+    DataStorageType,
+    read_file_directly,
+    write_file_directly,
+)
+
+
+class InvalidTargetError(ValueError):
+    ...
+
+
+class TargetIdentifiers(DataModelAbstractBase):
+    """
+    Identifiers for a Ligand
+    """
+
+    target_type: Optional[TargetType] = Field(
+        None,
+        description="Dynamic Enum describing the target type e.g sars2, mers or mac1",
+    )
+
+    fragalysis_id: Optional[str] = Field(
+        None, description="The PDB code of the target if applicable"
+    )
+
+    pdb_code: Optional[str] = Field(
+        None, description="The PDB code of the target if applicable"
+    )
+
+
+class Target(DataModelAbstractBase):
+    """
+    Schema for a Target
+    """
+
+    target_name: str = Field(None, description="The name of the target")
+
+    ids: Optional[TargetIdentifiers] = Field(
+        None,
+        description="TargetIdentifiers Schema for identifiers associated with this ligand",
+    )
+
+    data: str = Field(
+        "",
+        description="PDB file stored as a string to hold internal data state",
+        repr=False,
+    )
+    data_format: DataStorageType = Field(
+        DataStorageType.pdb,
+        description="Enum describing the data storage method",
+        allow_mutation=False,
+    )
```

Comment:
Same comment for ligand schema, do we want to make this a part of the base class?

Or I guess if we keep it manually consistent then that works I guess